### ISSUE TITLE
Extensions incorrectly reported as mandatory when constrained out of profile.

### ISSUE THREAD

**Issue by dxh73** [FirelyTeam/fhir-net-api#538]
A problem exists where an extension is being reported as included in an xml example structure when it is not included. For example:

Invalid	Instance count for 'Patient.extension:registrationDetails' is 1, which is not within the specified cardinality of 0..0

XML example:

``` xml
<Patient>
				<id value="dffd3330-dc21-11e7-9296-cec278b6b50a"/>
				<meta>
					<profile value="https://fhir.nhs.uk/STU3/StructureDefinition/CareConnect-EMS-PDS-Baby-Patient-1"/>
				</meta>
				<extension url="https://fhir.hl7.org.uk/STU3/StructureDefinition/Extension-CareConnect-EthnicCategory-1">
					<valueCodeableConcept>
						<coding>
							<system value="https://fhir.hl7.org.uk/STU3/CareConnect-EthnicCategory-1"/>
							<code value="A"/>
							<display value="British, Mixed British"/>
						</coding>
					</valueCodeableConcept>
				</extension>
				<identifier>
					<extension url="https://fhir.hl7.org.uk/STU3/StructureDefinition/Extension-CareConnect-NHSNumberVerificationStatus-1">
						<valueCodeableConcept>
							<coding>
								<system value="https://fhir.hl7.org.uk/STU3/CareConnect-NHSNumberVerificationStatus-1"/>
								<code value="01"/>
								<display value="Number present and verified"/>
							</coding>
						</valueCodeableConcept>
					</extension>
					<system value="https://fhir.nhs.uk/Id/nhs-number"/>
					<value value="9912003888"/>
				</identifier>
				<name>
					<use value="usual"/>
					<family value="DAWKINS"/>
					<given value="Jack"/>
					<period >
						<start value="2017-10-02"/>
					</period>
				</name>
				<gender value="male"/>
				<birthDate value="2017-10-02">
					<extension url="http://hl7.org/fhir/StructureDefinition/patient-birthTime">
						<valueDateTime value="2017-10-02T20:12:00+00:00"/>
					</extension>
				</birthDate>
				<address>
					<use value="home"/>
					<line value="4 Sandmoor Drive"/>
					<city value="LEEDS"/>
					<postalCode value="LS17 7DF"/>
				</address>
				<multipleBirthInteger value="1"/>
			</Patient>
```

The Patient.extension:registrationDetails extension is not in the example being validated. This extension is not mandatory (its profiled as 0..0).

If no extensions are included in an example, this error is not presented.

**Comment by marcovisserFurore** [comment]
@dxh73: can you give me the profile https://fhir.nhs.uk/STU3/StructureDefinition/CareConnect-EMS-PDS-Baby-Patient-1. Without this profile I cannot reproduce the issue.

**Comment by bushra** [comment]
As requested: 

[CareConnect-EMS-PDS-Baby-Patient-1.zip](https://github.com/ewoutkramer/fhir-net-api/files/1991287/CareConnect-EMS-PDS-Baby-Patient-1.zip)

**Comment by marcovisserFurore** [comment]
We are cleaning up old issues, and we noticed this issue was still open. Our apologies that it took so long that we gave a feedback on this issue.

Today I investigated the Patient instance above and did a validation against your profile (https://fhir.nhs.uk/STU3/StructureDefinition/CareConnect-EMS-PDS-Baby-Patient-1). It gave back an error about the Patient.Name. It lacks the use of `official`, which is correct, because the instance only has `usual` as use. Changing this `use` attribute to `official` the validator reports a success.

There is no errormessage anymore about the extensions, so I assume that has been solved. Therefor I close this issue.
